pipx/commands.py

```diff
@@ -453,10 +453,22 @@ def uninstall(venv_dir: Path, package: str, local_bin_dir: Path, verbose: bool):
 
     venv = Venv(venv_dir, verbose=verbose)
 
-    metadata = venv.get_venv_metadata_for_package(package)
-    app_paths = metadata.app_paths
-    for dep_paths in metadata.app_paths_of_dependencies.values():
-        app_paths += dep_paths
+    if venv.python_path.is_file():
+        # has a valid python interpreter and can get metadata about the package
+        metadata = venv.get_venv_metadata_for_package(package)
+        app_paths = metadata.app_paths
+        for dep_paths in metadata.app_paths_of_dependencies.values():
+            app_paths += dep_paths
+    else:
+        # doesn't have a valid python interpreter. We'll take our best guess on what to uninstall
+        # here.
+        apps_linking_to_venv_bin_dir = [
+            f
+            for f in constants.LOCAL_BIN_DIR.iterdir()
+            if str(f.resolve()).startswith(str(venv.bin_path))
```

Comment:
I was thinking maybe you could check `f.resolve().relative_to(venv.bin_path)` instead of string operations, but this might be too clunky since it throws a `ValueError` if it's not relative the other path.

Reply:
> package black has invalid interpreter /home/csmith/Downloads/Python-3.8.0b3/python3.8

Should we suggest `reinstall-all` here in the output or would that be too spammy?

Also, a bit of a side trek but would it be worth introducing `reinstall` for an individual package?  I could possible see a need if only certain venvs were made with a Python interpreter that is no longer present but it may be a bit of a niche case.  The instance I've run into this when I've had multiple Python versions compiled via Pyenv.

Reply:
I was thinking about that too. Created https://github.com/pipxproject/pipx/issues/256.

Reply:
> I was thinking maybe you could check f.resolve().relative_to(venv.bin_path) instead of string operations, but this might be too clunky since it throws a ValueError if it's not relative the other path.

That is a good suggestion. I usually go with path operations, but I feel like the current string matching is simpler.

Now that I think about it, this won't work for Windows since pipx doesn't use symlinks on Windows, so I will have to add a check for that.